docs/developer-guide/working-with-plugins.md

```diff
@@ -66,6 +66,28 @@ module.exports = {
 };
 ```
 
+### Custom Environments in Plugins
+
+Plugins can also provide custom environments. Custom environments can declare `globals`, turn `ecmaFeatures` on and off,
+and configure both core rules as well as rules included in the plugin. To provide custom environment, include `environment`
+property in your exported object.
+
+```js
+module.exports = {
+    environments: {
+        "my-custom-environment": {
+            globals: {
+                "global-variable": true
+            },
+            rules: {
+                "no-alert": 0, //turn off no-alert rule
+                "my-plugin/my-rule": 1 //turn on custom rule from this plugin (given that plugin's name is "my-plugin")
```

Comment:
It's a third way only inside of the plugin - outside there are still just two ways.

As I mentioned, I'm just not thrilled with people needing to use their own plugin name inside the plugin. The name of the plugin could change and they'd need to hunt down those references, which is exactly why rules are exported without the plugin name. I'm open to other suggestions.